frontend/src/metabase/core/components/AutocompleteInput/AutocompleteInput.tsx

```diff
@@ -5,45 +5,73 @@ import TextInput from "metabase/components/TextInput";
 import { TextInputProps } from "metabase/components/TextInput/TextInput";
 
 import { composeEventHandlers } from "metabase/lib/compose-event-handlers";
+import { useListKeyboardNavigation } from "metabase/hooks/use-list-keyboard-navigation";
+
 import { OptionsList } from "./AutocompleteInput.styled";
 
 interface AutocompleteInputProps extends TextInputProps {
   options?: string[];
+  filterFn?: (value: string | undefined, options: string[]) => string[];
+  onOptionClick?: (value: string) => void;
 }
 
+const DeafultFilterFn = (value: string | undefined, options: string[]) => {
+  if (!value || value.length === 0) {
+    return options;
+  }
+
+  return options.filter(option => {
+    const optionLowerCase = option.toLowerCase().trim();
+    const valueLowerCase = value.toLowerCase().trim();
+    return (
+      optionLowerCase.includes(valueLowerCase) &&
+      !(optionLowerCase === valueLowerCase)
+    );
+  });
+};
+
 const AutocompleteInput = ({
   value,
   onChange,
   options = [],
+  filterFn = DeafultFilterFn,
   onBlur,
+  onOptionClick,
```

Comment:
nit: should it be `onOptionSelect`? It is called not only for clicks, but also when an option is selected via keyboard